src/sbp_utils.c

```diff
@@ -188,6 +188,30 @@ void sbp_make_heading(msg_baseline_heading_t *baseline_heading, const gps_time_t
     baseline_heading->flags = flags;
 }
 
+u32 min_lock_time_decode(u8 lock_time_packed) {
+  if (lock_time_packed == 0) {
+    return 0;
+  }
+  else {
+  return 1 << (lock_time_packed + 4);
+  }
+}
+
+u8 lock_time_encode(u32 lock_time) {
+  if (lock_time < 32)
+     return 0;
+   else {
+       for(u8 i = 0; i < 16; i++) {
+         if ((int)lock_time > (1<<(i+5))) {
+           continue;
+         }
+         else
+           return i;
+        }
+     }
+     return 15;
+}
```

Comment:
Style is off on these two (indentation, curly braces). Logic could also be simplified a little bit:

```c
u32 min_lock_time_decode(u8 lock_time_packed) {
  if (lock_time_packed == 0)
    return 0;

  return 1 << (lock_time_packed + 4);
}

u8 lock_time_encode(u32 lock_time) {
  for(u8 i = 0; i < 16; i++)
    if (lock_time <= (u32)(1 << (i+5)))
      return i;

  return 15;
}
```

Would be good to add some unit tests for these as well.